WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Package Details/Package Selection/Package Creation/ShippingLabelCustomPackageFormViewModel.swift

```diff
@@ -13,26 +13,65 @@ final class ShippingLabelCustomPackageFormViewModel: ObservableObject {
 
     /// The name of the custom package
     ///
-    @Published var packageName: String = ""
+    @Published var packageName: String
+
+    /// The type of the custom package, set to `box` by default
+    ///
+    @Published var packageType: PackageType
 
     /// The length of the custom package
     ///
-    @Published var packageLength: String = "0"
+    @Published var packageLength: String
 
     /// The width of the custom package
     ///
-    @Published var packageWidth: String = "0"
+    @Published var packageWidth: String
 
     /// The height of the custom package
     ///
-    @Published var packageHeight: String = "0"
+    @Published var packageHeight: String
 
     /// The weight of the custom package when empty
     ///
-    @Published var emptyPackageWeight: String = "0"
+    @Published var emptyPackageWeight: String
+
+    init(lengthUnit: String? = ServiceLocator.shippingSettingsService.dimensionUnit,
+         weightUnit: String? = ServiceLocator.shippingSettingsService.weightUnit,
+         packageName: String = "",
+         packageType: PackageType = .box,
+         packageLength: String = "",
+         packageWidth: String = "",
+         packageHeight: String = "",
+         emptyPackageWeight: String = "") {
```

Comment:
It's not apparent to me why we need to keep all these in the intializer. Maybe I'm missing out on some special cases - can you help me understand here?

Reply:
My thinking here was to keep them in the initializer because it makes it easy to initialize an empty form, but it also would make the form more flexible/reusable later on, for example if we want to support editing existing custom packages in the future. Do you think that's overkill here? (I'm still trying to figure out the right balance between implementations that make sense just for what we're doing now vs. thinking ahead to how we might want to use them later on.)

Reply:
> for example if we want to support editing existing custom packages in the future

Oh I get it now! I see that we have that edit feature in the web also so that makes sense to think a bit ahead! Although I'd prefer to keep the initializer "neat" with just an optional package (which defaults to `nil`) and we can use it to fill out the fields. That also matches with what we're doing with other view models so far.

But if we allow editing then we'll need to create the view model from outside and inject it into the view instead of letting the view create it like what we're having now.

Reply:
> Although I'd prefer to keep the initializer "neat" with just an optional package (which defaults to nil) and we can use it to fill out the fields. That also matches with what we're doing with other view models so far.

Ah, that's definitely a simpler approach. Thanks for the suggestion! I'll make that change in my next PR. (It's true that we'll still need to make changes to how the view model is created, to support that scenario, but at least the view model will be set up for it already.)